src/gui/guiEngine.cpp

```diff
@@ -435,9 +435,21 @@ void GUIEngine::drawBackground(video::IVideoDriver *driver)
 		return;
 	}
 
+	v2u32 screensize_orig = screensize;
+	v2f32 scale(
+			(f32) screensize.X / sourcesize.X,
+			(f32) screensize.Y / sourcesize.Y);
+	if (scale.X < scale.Y)
+		screensize.X = (int) (scale.Y * sourcesize.X);
```

Comment:
better to write to a variable called `bg_size` or similar than to write to `screensize`.